lib/bolt/result.rb

```diff
@@ -7,47 +7,61 @@ module Bolt
   class Result
     attr_reader :target, :value, :action, :object
 
-    def self.from_exception(target, exception, action: 'action')
+    def self.from_exception(target, exception, action: 'action', position: [])
+      details = create_details(position)
       if exception.is_a?(Bolt::Error)
-        error = exception.to_h
+        error = { 'details' => details }.merge(exception.to_h)
```

Comment:
Should this be a deep merge? This will wipe out any existing details that the error might have already had.

Reply:
Yike yes good catch!